src/runtime/hexagon/hexagon/hexagon_device_api_v2.cc

```diff
@@ -148,20 +139,29 @@ void HexagonDeviceAPIv2::CopyDataFromTo(DLTensor* from, DLTensor* to, TVMStreamH
   CHECK_EQ(to->byte_offset, 0);
   CHECK_EQ(GetDataSize(*from), GetDataSize(*to));
 
-  HexagonBuffer* hex_from_buf = static_cast<HexagonBuffer*>(from->data);
-  HexagonBuffer* hex_to_buf = static_cast<HexagonBuffer*>(to->data);
+  auto lookup_hexagon_buffer = [this](void* ptr) -> HexagonBuffer* {
+    auto it = this->hexagon_buffer_map_.find(ptr);
+    if (it != this->hexagon_buffer_map_.end()) {
+      return it->second.get();
+    }
+    return nullptr;
```

Comment:
Since the `nullptr` check is done after each lookup, let's move it in here as `CHECK(it != this->hexagon_buffer_map_.end())`.

Reply:
Agree with @Lunderberg's comment and also think that `lookup_hexagon_buffer` could / should be a member of `HexagonDeviceAPIv2` named `LookupHexagonBuffer` alongside the "Allocate" and "Free" methods.

Reply:
I like that, because that would let it be used in the `free_nd` function as well.  Though, I think it should be private, unlike the "Allocate" and "Free" methods.